File: Frontend/src/Models/Owner.js
import Person from "./Person";

class Owner extends Person{
    constructor({id, civility, sex, firstName, lastName, address, zipCode, city, country, birthDate, age, placeOfBirth, nationalRegister, email, phoneNumber}) {
        super(id,civility, phoneNumber, email, address, city, country, firstName, zipCode, sex, lastName);
        this.placeOfBirth=placeOfBirth;
        this.age=age;
        this.nationalRegister=nationalRegister;
        this.birthDate=birthDate;
    }
}
export default Owner

File: Backend/Models/Owner.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models
{
    public class Owner
    {
        [Key]
        public int id { get; set; }
        [Required]
        public string civility { get; set; }
        [Required]
        public string gender { get; set; }
        [Required]
        public string lastName { get; set; }
        [Required]
        public string firstName { get; set; }
        [Required]
        public string address { get; set; }
        [Required]
        public int zipCode { get; set; }
        [Required]
        public string city { get; set; }
        [Required]
        public string country { get; set; }
        public string birthDate { get; set; }
        public int age { get; set; }
        public string placeOfBirth { get; set; }
        public string nationalRegister { get; set; }
        [Required]
        public string email { get; set; }
        [Required]
        public string phoneNumber { get; set; }
        public int ownerInCharge { get; set; }
        
    }
}
